tests/test_geo_utils.py

```diff
@@ -5,58 +5,70 @@
 
 import pytest
 
-from sentinelhub import CRS, BBox, geo_utils
+from sentinelhub import CRS, BBox
+from sentinelhub.geo_utils import (
+    bbox_to_dimensions,
+    bbox_to_resolution,
+    get_image_dimension,
+    get_utm_crs,
+    pixel_to_utm,
+    to_wgs84,
+    transform_point,
+    utm_to_pixel,
+    wgs84_to_pixel,
+    wgs84_to_utm,
+)
 
+BBOX_WGS84 = BBox(((111.644, 8.655), (111.7, 8.688)), CRS.WGS84)
+BBOX_UTM = BBox(((570851.8316965176, 956770.2133183606), (577006.4223865959, 960429.6742984429)), CRS("32649"))
+BBOX_POP_WEB = BBox(((12428153.230124235, 967155.4076988235), (12434387.12160866, 970871.4287682716)), CRS.POP_WEB)
 
-def test_wgs84_to_utm() -> None:
-    x, y = geo_utils.wgs84_to_utm(15.525078, 44.1440478, CRS.UTM_33N)
-    assert (x, y) == pytest.approx((541995.694062, 4888006.132887), rel=1e-8)
 
+@pytest.mark.parametrize("input_bbox, expected_bbox", [(BBOX_WGS84, BBOX_UTM), (BBOX_WGS84, BBOX_POP_WEB)])
+def test_wgs84_to_utm(input_bbox: BBox, expected_bbox: BBox) -> None:
+    point_lower_left = wgs84_to_utm(*input_bbox.lower_left, expected_bbox.crs)
+    point_upper_right = wgs84_to_utm(*input_bbox.upper_right, expected_bbox.crs)
 
-def test_to_wgs84() -> None:
-    lng, lat = geo_utils.to_wgs84(541995.694062, 4888006.132887, CRS.UTM_33N)
-    assert (lng, lat) == pytest.approx((15.525078, 44.1440478), rel=1e-8)
+    assert point_lower_left == pytest.approx(expected_bbox.lower_left, rel=1e-8)
+    assert point_upper_right == pytest.approx(expected_bbox.upper_right, rel=1e-8)
 
 
-def test_get_utm_crs() -> None:
-    lng, lat = 15.52, 44.14
-    crs = geo_utils.get_utm_crs(lng, lat)
-    assert crs is CRS.UTM_33N
+@pytest.mark.parametrize("input_bbox, expected_bbox", [(BBOX_UTM, BBOX_WGS84), (BBOX_POP_WEB, BBOX_WGS84)])
+def test_to_wgs84(input_bbox: BBox, expected_bbox: BBox) -> None:
+    assert to_wgs84(*input_bbox.lower_left, input_bbox.crs) == pytest.approx(expected_bbox.lower_left, rel=1e-8)
+    assert to_wgs84(*input_bbox.upper_right, input_bbox.crs) == pytest.approx(expected_bbox.upper_right, rel=1e-8)
 
 
-def test_bbox_to_resolution() -> None:
-    bbox = BBox(((111.644, 8.655), (111.7, 8.688)), CRS.WGS84)
-    resx, resy = geo_utils.bbox_to_resolution(bbox, 512, 512)
+def test_get_utm_crs() -> None:
+    assert get_utm_crs(*BBOX_WGS84.lower_left) is BBOX_UTM.crs
+    assert get_utm_crs(*BBOX_WGS84.upper_right) is BBOX_UTM.crs
+
 
-    assert (resx, resy) == pytest.approx((12.0207, 7.1474), rel=1e-4)
+@pytest.mark.parametrize("input_bbox", [BBOX_WGS84, BBOX_UTM, BBOX_POP_WEB])
+def test_bbox_to_resolution(input_bbox: BBox) -> None:
+    assert bbox_to_resolution(input_bbox, 512, 512) == pytest.approx((12.0207, 7.1474), rel=1e-4)
```

Comment:
you're still just testing just a single bbox right? just in different CRS

wouldn't it be better to have three different bboxes to test all this?